pandas/io/parsers.py

```diff
@@ -3146,11 +3148,11 @@ def converter(*date_cols):
                 return tools.to_datetime(
                     ensure_object(strs),
                     utc=None,
-                    box=False,
                     dayfirst=dayfirst,
                     errors='ignore',
                     infer_datetime_format=infer_datetime_format
-                )
+                ).to_numpy()
```

Comment:
do things break if you don't .to_numpy() ?

Reply:
They do, in at least one place, though also in an internal function. `sanitize_objects` in _libs/parsers.pyx requires an ndarray in the first argument. Without `.numpy()` here it gets an Index.

Reply:
Actually, I'm not sure we can get away without this here.  The result eventually ends up as an argument to the initializer for `Block` which uses it for `self.values.  

So we can move the conversion of the index to `ndarray`, but it has to happen eventually.  If we really want to move it out of here, then perhaps we can move the conversion into `Block`. Check there for the type of the incoming `values` and convert to `ndarray` if necessary.